backend/src/cards/deck.rs

```diff
@@ -0,0 +1,76 @@
+use crate::cards::card::{Card, CardColor, CardSymbol};
+use rand::seq::SliceRandom;
+
+pub struct Deck {
+    draw_pile: Vec<Card>,
+    discard_pile: Vec<Card>,
+}
+
+impl Deck {
+    pub fn new() -> Deck {
+        let mut draw_pile = Vec::new();
+
+        for color in CardColor::non_black_iter() {
+            for parity in 0..2 {
+                for value in 0..=9 {
+                    if parity == 0 && value == 0 {
+                        // only one value=0 card of each color
+                        continue;
+                    }
+
+                    draw_pile.push(Card::new(color, CardSymbol::Value(value)).unwrap())
+                }
+
+                for symbol in [CardSymbol::Skip, CardSymbol::Reverse, CardSymbol::Draw2] {
+                    draw_pile.push(Card::new(color, symbol).unwrap())
+                }
+
+                draw_pile.push(
+                    Card::new(
+                        CardColor::Black,
+                        if parity == 0 {
+                            CardSymbol::Wild
+                        } else {
+                            CardSymbol::Draw4
+                        },
+                    )
+                    .unwrap(),
+                )
+            }
+        }
```

Comment:
Za mě jsou asi lepší hloupá, ale dobře čitelná řešení, význam `parity` mi chvíli trval pochopit :) Ale tady je to ještě v pohodě, takže to tak klidně můžeš nechat

Reply:
Yeah, I didn't know what else to call it... but you get 2 of most cards, exception being 0 and wild/draw4, so it made sense to have the 0..2 for-loop.

Reply:
The clean solution (in my opinion, as these things are quite subjective) would be to break it down into several functions (e.g. `insert_black_cards()` and `insert_colour_cards()`), maybe use `.map()` for colours, get rid of parity.

Generally you want to write code that can be easily modified in order to satisfy a sudden change in the task without having to rewrite many lines. E.g. we could want to have a few more wildcards in the deck.

The logic of creating a deck of cards is relatively simple and cleaning up is not really necessary here, but having a similarly complex logic for a non-trivial feature could be devastating if a requirement changes - and in practice, they change quite a lot :)

Reply:
Amended into f0a1c0e, force-pushed.